test/common/http/conn_manager_impl_test.cc

```diff
@@ -2048,6 +2048,61 @@ TEST_F(HttpConnectionManagerImplTest, TestStreamIdleAccessLog) {
   EXPECT_EQ(1U, stats_.named_.downstream_rq_idle_timeout_.value());
 }
 
+// Test timeout when route timeout needs to be used if max_stream_duration is not specified.
+// Regression test for https://github.com/envoyproxy/envoy/issues/15530.
+// TODO(ramaraochavali): move this test to DurationTimeout test.
+TEST_F(HttpConnectionManagerImplTest, DurationTimeoutFromRouteTimeout) {
```

Comment:
I had to create a separate test for this because `clusterInfo` is repeatedly called and it keeps triggering the clearRouteCache so the timer gets fired multiple times. The same problem exists for https://github.com/envoyproxy/envoy/blob/main/test/common/http/conn_manager_impl_test.cc#L2173 where timer is repeatedly enabled. Having two such tests in the same test is not working. So I made it as separate test and TODO to look at the issue later.